src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
-export interface Figure {
+enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Colors {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea: () => number;
 }
 
 export class Triangle {
+  shape = Shapes.Triangle;
+
+  constructor(
+    public color: Colors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
Create a helper function that accepts any number of args and checks them. Reuse it with all shapes.